sensors/triggers/slack/slack.go

```diff
@@ -0,0 +1,139 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package slack
+
+import (
+	//"bytes"
+	"encoding/json"
+	"github.com/argoproj/argo-events/common"
+	"k8s.io/client-go/kubernetes"
+
+	//"github.com/argoproj/argo-events/common"
+	"net/http"
+
+	"github.com/argoproj/argo-events/pkg/apis/sensor/v1alpha1"
+	"github.com/argoproj/argo-events/sensors/triggers"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	//offical:
+	//"github.com/slack-go/slack"
+
+	//old:
+	"github.com/nlopes/slack"
+)
+
+type SlackTrigger struct {
+	// K8sClient is the Kubernetes client
+	K8sClient kubernetes.Interface
+	// Sensor refer to the sensor object
+	Sensor *v1alpha1.Sensor
+	// Trigger refers to the trigger resource
+	Trigger *v1alpha1.Trigger
+	// Logger to log stuff
+	Logger *logrus.Logger
+	// http client to invoke function.
+	httpClient *http.Client
+}
+
+// NewSlackTrigger returns a new Slack trigger context
+func NewSlackTrigger(k8sClient kubernetes.Interface, sensor *v1alpha1.Sensor, trigger *v1alpha1.Trigger, logger *logrus.Logger, httpClient *http.Client) (*SlackTrigger, error) {
+	return &SlackTrigger{
+		K8sClient:  k8sClient,
+		Sensor:     sensor,
+		Trigger:    trigger,
+		Logger:     logger,
+		httpClient: httpClient,
+	}, nil
+}
+
+func (t *SlackTrigger) FetchResource() (interface{}, error) {
+	return t.Trigger.Template.Slack, nil
+}
+
+func (t *SlackTrigger) ApplyResourceParameters(sensor *v1alpha1.Sensor, resource interface{}) (interface{}, error) {
+	resourceBytes, err := json.Marshal(resource)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal the Slack trigger resource")
+	}
+	parameters := t.Trigger.Template.Slack.Parameters
+
+	if parameters != nil {
+		updatedResourceBytes, err := triggers.ApplyParams(resourceBytes, t.Trigger.Template.Slack.Parameters, triggers.ExtractEvents(sensor, parameters))
+		if err != nil {
+			return nil, err
+		}
+
+		var st *v1alpha1.SlackTrigger
+		if err := json.Unmarshal(updatedResourceBytes, &st); err != nil {
+			return nil, errors.Wrap(err, "failed to unmarshal the updated Slack trigger resource after applying resource parameters")
+		}
+
+		return st, nil
+	}
+
+	return resource, nil
+}
+
+
+// Execute executes the trigger
+func (t *SlackTrigger) Execute(resource interface{}) (interface{}, error) {
+	t.Logger.Infoln("executing SlackTrigger")
+	_, ok := resource.(*v1alpha1.SlackTrigger)
+	if !ok {
+		return nil, errors.New("failed to marshal the Slack trigger resource")
+	}
+
+	slacktrigger := t.Trigger.Template.Slack
+
+	channel := slacktrigger.Channel
+	if channel == "" {
+		return nil, errors.New("no slack channel provided")
+	}
+
+	message := slacktrigger.Message
+	if message == "" {
+		return nil, errors.New("no slack message to post")
+	}
+
+	slackToken := ""
+	if slacktrigger.SlackToken != nil {
+		secretToken, err := common.GetSecrets(t.K8sClient, slacktrigger.Namespace, slacktrigger.SlackToken)
```

Comment:
remove `secretToken` variable, just assign `slackToken` the value of secret data,

`slackToken, err := common.GetSecrets(t.K8sClient, slacktrigger.Namespace, slacktrigger.SlackToken)`